primary/tests/nodes_bootstrapping_tests.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use itertools::Itertools;
+use std::{collections::HashMap, time::Duration};
+use telemetry_subscribers::TelemetryGuards;
+use test_utils::cluster::Cluster;
+use tracing::info;
+
+/// Nodes will be started in a staggered fashion. This is simulating
+/// a real world scenario where nodes across validators will not start
+/// in the same time.
+#[ignore]
+#[tokio::test]
+async fn test_node_staggered_starts() {
+    // Enabled debug tracing so we can easily observe the
+    // nodes logs.
+    let _guard = setup_tracing();
+
+    let node_staggered_delay = Duration::from_secs(60 * 5); // 5 minutes
+
+    // A cluster of 4 nodes will be created
+    let cluster = Cluster::new(None, None);
+
+    // ==== Start first authority ====
+    cluster.authority(0).start(false, Some(1)).await;
+
+    tokio::time::sleep(node_staggered_delay).await;
+
+    let rounds = authorities_current_round(&cluster).await;
+    assert_eq!(
+        rounds.len(),
+        1,
+        "Expected to have received metric only from one node"
+    );
+    assert!(rounds.values().all(|v| v == &1.0), "We have (f+2) unavailable nodes and expected all nodes to have made progress up to round 1 only");
+
+    // ==== Start second authority ====
+    cluster.authority(1).start(false, Some(1)).await;
+
+    tokio::time::sleep(node_staggered_delay).await;
+
+    let rounds = authorities_current_round(&cluster).await;
+    assert_eq!(
+        rounds.len(),
+        2,
+        "Expected to have received metrics from only two nodes"
+    );
+    assert!(rounds.values().all(|v| v == &1.0), "We have (f+1) unavailable nodes and expected all nodes to have made progress up to round 1 only");
+
+    // ==== Start third authority ====
+    // Now 2f + 1 nodes are becoming available and we expect all the nodes to
+    // start making progress (advance in rounds).
+    cluster.authority(2).start(false, Some(1)).await;
+
+    tokio::time::sleep(node_staggered_delay).await;
+
+    let rounds = authorities_current_round(&cluster).await;
+    assert_eq!(
+        rounds.len(),
+        3,
+        "Expected to have received metrics from only three nodes"
+    );
+    assert!(rounds.values().all(|v| v > &1.0), "We have only (f) unavailable nodes, so all should have made progress more than just the first round");
+
+    // We expect all the nodes to have managed to catch up by now
+    // and be pretty much in similar rounds. The threshold here is
+    // not defined by some rule but more of an expectation.
+    let (min, max) = rounds.values().into_iter().minmax().into_option().unwrap();
+    assert!(max - min <= 2.0, "Nodes shouldn't be that behind");
+
```

Comment:
Here, we want to witness some commits coming out of consensus! What we're looking for is e.g. 
```bash
2022-07-22T13:36:39.591669Z DEBUG consensus::bullshark: Processing y7GMf2mjXh9vxHPD: C345(IP26ybELdYe7p7W8FjvOaeeW1x5O1EwQ/LRIhon3oUQ=, YJjFEbQw7FwE5HUI, E0)
2022-07-22T13:36:39.591825Z DEBUG consensus::bullshark: Leader /kJ8cPd9/pDsr8l/: C344(IP26ybELdYe7p7W8FjvOaeeW1x5O1EwQ/LRIhon3oUQ=, G0lhHhA9zsHbXOA0, E0) has enough support
2022-07-22T13:36:39.591960Z DEBUG consensus::utils: Processing sub-dag of zG8IWEouTkMF9rPy: C343(vq2gYSbHjZi0oaafbuYYlpTw9HUVONqCTxrcixShtWI=, 6CJnWcb+KE/5KwxQ, E0)
2022-07-22T13:36:39.592035Z DEBUG consensus::utils: Sequencing zG8IWEouTkMF9rPy: C343(vq2gYSbHjZi0oaafbuYYlpTw9HUVONqCTxrcixShtWI=, 6CJnWcb+KE/5KwxQ, E0)
2022-07-22T13:36:39.592149Z DEBUG consensus::utils: Sequencing rKHidx0qFCTtpdTi: C342(IP26ybELdYe7p7W8FjvOaeeW1x5O1EwQ/LRIhon3oUQ=, C02GTrIzhxN2ZAVe, E0)
2022-07-22T13:36:39.592264Z DEBUG consensus::utils: Sequencing zZR2+G3R5Zwci32J: C341(IP26ybELdYe7p7W8FjvOaeeW1x5O1EwQ/LRIhon3oUQ=, SNYDJLc2/oKxY0zE, E0)
2022-07-22T13:36:39.592377Z DEBUG consensus::utils: Sequencing +yxztL/krEIsUhFy: C340(vq2gYSbHjZi0oaafbuYYlpTw9HUVONqCTxrcixShtWI=, COtpuI4D2gCPYP54, E0)
2022-07-22T13:36:39.592480Z DEBUG consensus::utils: Sequencing 7b8G3O7J59NddTjg: C340(deQXTdWIIlSAhvF7A3zssO6GUWt9E0AKgMhWtL2vf+E=, gVzH2bA0ZmoYkC/g, E0)
2022-07-22T13:36:39.592586Z DEBUG consensus::utils: Sequencing qH4zkHUDvehcg8Y+: C341(YxwVQfOkv0TU2JcGFWSqhJXXZvYZGj/2FWIAPxhLjGU=, 7XjsbTteV41tjjMx, E0)
2022-07-22T13:36:39.592689Z DEBUG consensus::utils: Sequencing GkBOnWDV7ZxJemO9: C341(deQXTdWIIlSAhvF7A3zssO6GUWt9E0AKgMhWtL2vf+E=, m/PAcyilLCaDAEjC, E0)
2022-07-22T13:36:39.592791Z DEBUG consensus::utils: Sequencing nXQVF4iy31tb7WeE: C342(YxwVQfOkv0TU2JcGFWSqhJXXZvYZGj/2FWIAPxhLjGU=, MGCyNxxWgpxslSDI, E0)
2022-07-22T13:36:39.592892Z DEBUG consensus::utils: Sequencing d1S62fsnw9YH4hT2: C342(vq2gYSbHjZi0oaafbuYYlpTw9HUVONqCTxrcixShtWI=, Mag9C7VUOy6/isi8, E0)
2022-07-22T13:36:39.594734Z DEBUG consensus::utils: Processing sub-dag of /kJ8cPd9/pDsr8l/: C344(IP26ybELdYe7p7W8FjvOaeeW1x5O1EwQ/LRIhon3oUQ=, G0lhHhA9zsHbXOA0, E0)
2022-07-22T13:36:39.594806Z DEBUG consensus::utils: Sequencing /kJ8cPd9/pDsr8l/: C344(IP26ybELdYe7p7W8FjvOaeeW1x5O1EwQ/LRIhon3oUQ=, G0lhHhA9zsHbXOA0, E0)
2022-07-22T13:36:39.594913Z DEBUG consensus::utils: Sequencing ylvRzLlMFCTRl7lz: C343(deQXTdWIIlSAhvF7A3zssO6GUWt9E0AKgMhWtL2vf+E=, sDbNGWvjAunRsQYz, E0)
2022-07-22T13:36:39.595010Z DEBUG consensus::utils: Sequencing mdMEkejOt7WdHeI+: C343(YxwVQfOkv0TU2JcGFWSqhJXXZvYZGj/2FWIAPxhLjGU=, TgajZzgAqm2XiKBd, E0)
2022-07-22T13:36:39.595675Z DEBUG consensus::bullshark: Latest commit of IP26ybELdYe7p7W8FjvOaeeW1x5O1EwQ/LRIhon3oUQ=: Round 344
2022-07-22T13:36:39.595715Z DEBUG consensus::bullshark: Latest commit of YxwVQfOkv0TU2JcGFWSqhJXXZvYZGj/2FWIAPxhLjGU=: Round 343
2022-07-22T13:36:39.595741Z DEBUG consensus::bullshark: Latest commit of vq2gYSbHjZi0oaafbuYYlpTw9HUVONqCTxrcixShtWI=: Round 343
2022-07-22T13:36:39.595763Z DEBUG consensus::bullshark: Latest commit of deQXTdWIIlSAhvF7A3zssO6GUWt9E0AKgMhWtL2vf+E=: Round 343
```
The rounds advancing are by themselves not proof of the network making progress (as opposed to nodes making progress).